### DISCUSSION TITLE
Notification /greyed out project presentation when project initialization fails

### DISCUSSION THREAD

**Discussion by hubertp** [enso-org/enso#7169] [Work In Progress - GUI]
(apologies if this was discussed before)

An unknown failure during initialization process, be it due to the engine/LS/GUI, can be rather frustrating.
But what I think is more frustrating, especially for new comers, is when the failure happens silently and we are left in a limbo state.
Consider a situation when something failed:
![Screenshot from 2023-06-29 20-00-39](https://github.com/enso-org/enso/assets/292128/a92e6024-db7d-4d1e-9af5-69959d9f82db)

Unless I look into logs (backend or console logs) I have no way of knowing if something is processing or not. Sometimes it is really doing something, because the machine might be less powerful. But oftentimes IDE very well knows that something failed and nothing will happen.
Users will not look into any failure unless told to, they will simply close the window and delete the program.

In this case I can see in the logs
```
router.ts:48 [ERROR] app/gui/src/presenter.rs:139 Could not create new project: Peer has replied with an error: Error { code: 4012, message: "Cannot find release `2023.2.1-dev`.", data: None }..
```
(it's my mistake, so nothing to investigate) 
It's just one of the many calls that can fail during initialization which IDE is perfectly aware of because it gets the response.

What I imagine would help is once a message like that is received, we should just block the whole UI (maybe grey out the whole presentation to make it even more visible) with a simple message to a) restart b) collect the logs and send them to us, if necessary.

Note that I'm not discussing the fact that errors shouldn't happen. That goes without saying but it is also a wishful thinking if we don't control where Enso is installed.

**Comment by farmaazon** [comment]
Well, this is a part of the #7163 - Before regression, we indeed were showing the error to the user.

Also, we plan to give even more information once the message board will be improved: #5201 I planned to do this for Beta release, but suddenly the CB redesign happened...

**Reply by hubertp** [reply]
@farmaazon I see, a notification. How about blocking the presentation as well? With initialization failure we know that the setup is not usable at all. And even if not initialized, one can still click around, try (and fail) to create nodes, partially open CB and so on. On such failure it looks to me like it is in a half-broken state and I'd prefer it looked either broken or working but not something in between.

**Reply by farmaazon** [reply]
You are right. I've updated the description of the task, as we already have "read only" flag, so it should be easy to just turn it on after failure.
